### PR TITLE
PMM-12686 Basic/Token auth between server and client.

### PR DESCRIPTION
**Pull request by JiriCtvrtka**
PMM-12686

Link to the Feature Build: SUBMODULES-0

### CODE REVIEW

**Comment by JiriCtvrtka** [comment]
Managed check fails because it needs modified grafana to pass.

File: agent/client/client.go [hunk 1]
```diff
@@ -842,6 +844,17 @@ func dial(dialCtx context.Context, cfg *config.Config, l *logrus.Entry) (*dialRe
 	}, nil
 }
 
+func getAuthorizationFromUserAndPassword(username, password string) string {
```
Comment (BupycHuk) [inline agent/client/client.go:847]:
```suggestion
func authorizationFromUserAndPassword(username, password string) string {
```

File: managed/services/agents/registry.go [hunk 2]
```diff
@@ -208,7 +213,14 @@ func (r *Registry) register(stream agentpb.Agent_ConnectServer) (*pmmAgentInfo,
 	return agent, nil
 }
 
-func (r *Registry) authenticate(md *agentpb.AgentConnectMetadata, q *reform.Querier) (*models.Node, error) {
+func (r *Registry) authenticate(ctx context.Context, md *agentpb.AgentConnectMetadata, q *reform.Querier) (*models.Node, error) {
```
Comment (BupycHuk) [inline managed/services/agents/registry.go:216]: do we have any tests for this method? can we add a few unit tests?

**Review by BupycHuk — COMMENTED** [review]
Code overall looks good to me, just curious if we could do it simpler by passing Authorization header and adding some viewer/editor role into https://github.com/percona/pmm/pull/2852/files#diff-67ea296ada70fcfe84d6724b468b58e3c9d7db6ed5d92e6d75774366976f17edR44?

File: managed/services/agents/registry.go (hunk 2 above)
Comment (JiriCtvrtka) [reply managed/services/agents/registry.go:216]: OK I will add them.

File: managed/services/grafana/auth_server.go [hunk 3]
```diff
@@ -40,8 +40,7 @@ import (
 
 // rules maps original URL prefix to minimal required role.
 var rules = map[string]role{
-	// TODO https://jira.percona.com/browse/PMM-4420
-	"/agent.Agent/Connect": none,
+	"/agent.Agent/Connect": admin,
```
Comment (ademidoff) [inline managed/services/grafana/auth_server.go:43]: 👍

File: managed/services/grafana/auth_server.go [hunk 4]
```diff
@@ -498,22 +512,31 @@ func (s *AuthServer) authenticate(ctx context.Context, req *http.Request, l *log
 		return nil, nil
 	}
 
-	// Get authenticated user from Grafana
-	authUser, authErr := s.getAuthUser(ctx, req, l)
-	if authErr != nil {
-		return nil, authErr
+	var user *authUser
+	skipAuth := isLocalAgentConnection(req)
+	if skipAuth {
```
Comment (ademidoff) [inline managed/services/grafana/auth_server.go:517]:
```suggestion
	if isLocalAgentConnection(req) {
```

**Review by ademidoff — APPROVED** [review]
👍

**Review by BupycHuk — APPROVED** [review]

**Pull request merged**